### DISCUSSION TITLE
Tailwind CSS (from nuxt ui) used in a module is purged

### DISCUSSION THREAD

**Discussion by guirak** [nuxt/nuxt#25051] [Questions]
Hello 

I have created a module that uses Nuxt UI (I want to be able to reuse this module in several projects). 

In this module, I have created basic Vue components that uses Tailwind CSS. 

When testing this module in the playground, the tailwind css classes are not applied on my module components. 

If I use a tailwind CSS class in the page of the playground, then immediatly this used css class is applied to the module components too and still be applied event if I remove it in the page and I clear my module and playground generated files. 

 It seems the purge CSS is applied to the module ignoring the defined component. 
 
Is there a settings to add in the module.ts ? 

Thank you

**Comment by awais019** [comment]
@guirak Did you find any solution I'm facing same problem

**Comment by guirak** [comment]
Hi @awais019, 

Read this page [https://v3.tailwindcss.com/docs/content-configuration](https://v3.tailwindcss.com/docs/content-configuration).

Tailwind uses now the content configuration to scan all the strings in files and providing the class name it found. 

I have solved the problem but with a personal approach, I don't know if it's the good way or not but for me it's working well int Nuxt UI v2. 

In my module, I have created a tailwind.config.ts composable in the path **composables/config**  :


Next I use this file in other module or my app. 

On my app (tailwind.config.ts) : 

```ts
// import config from "@mmiUIFwk/config/tailwind.config";
import {mmioneErpUiFwkTailwindConfig} from "../../../../../ui/mmione-erp-ui-fwk/src/runtime/config/tailwind.config"

export default mmioneErpUiFwkTailwindConfig
```

On my module - level 1 (runtime/composables/config/tailwind.config.ts ) :

```ts
import type {Config} from 'tailwindcss'
import path from 'path';
import { merge } from 'lodash';

import {webuiTailwindConfig} from "../../../../../ui/mmione-webui-fwk/src/runtime/config/tailwind.config"

export const mmioneErpUiFwkTailwindConfig: Partial<Config> = merge({}, webuiTailwindConfig, {

  content: [
    ...webuiTailwindConfig.content,
    path.resolve(__dirname, "../components/**/*.{vue,mjs,ts}"),
    path.resolve(__dirname, "../composables/**/*.{vue,mjs,ts}"),
  ]
});

```

On my GUI framework module - level 2 (runtime/composables/config/tailwind.config.ts ): 


```ts

// @ts-ignore
import type {Config} from 'tailwindcss'
// @ts-ignore
import plugin from "tailwindcss/plugin";

import path from 'path';

export const webuiTailwindConfig: Partial<Config> = {
  theme: {
    extend: {
      colors: {
        ["san-marino"]: {
          50: "#f3f6fc",
          100: "#e6edf8",
          200: "#c7d8f0",
          300: "#96b8e3",
          400: "#5d93d3",
          500: "#346caf",
          600: "#285ca1",
          700: "#224a82",
          800: "#1f406d",
          900: "#1f375b",
          950: "#15233c"
        }
      }
    }
  },
  plugins: [
    // @ts-ignore
    plugin(function ({addVariant}) {
      addVariant('portrait', '@media (orientation: portrait)');
      addVariant('landscape', '@media (orientation: landscape)');
    })
  ],

  safelist: [
    {
      pattern: /(bg|text|ring)-(.*)-(50|100|200|300|400|500|600|700|800|900|950)/
    },
    {
      pattern: /(top|bottom|left|right)-(.*)/,
    },
    {
      pattern: /grid-cols-(1|2|3|4|5|6|7|8)/,
    },
    {
      pattern: /(col-span)-(.*)/,
    }
  ],

  content: [
    path.resolve(__dirname, "../components/**/*.{vue,mjs,ts}"),
    path.resolve(__dirname, "../config/**/*.{vue,mjs,ts}"),
  ],
}


```

**Comment by awais019** [comment]
@import "tailwindcss" theme(static) source("../../../..");
@import "@nuxt/ui";

@source "../../../content/**/*";
@source "../../../node_modules/.c12";

I found this solution from one of github issues. it works fine
